pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -285,7 +299,13 @@ func (h *handler) run(obj runtime.Object, create bool) (runtime.Object, error) {
 		return nil, err
 	}
 
-	if err := h.apply.WithOwner(obj).ApplyObjects(objs...); err != nil {
+	applier := h.apply.WithOwner(obj)
+	if !create {
+		// If the infrastructure is being deleted, ignore previously applied objects.
+		// If creation failed, this will allow deletion to process.
+		applier = applier.WithIgnorePreviousApplied()
+	}
+	if err := applier.ApplyObjects(objs...); err != nil {
```

Comment:
This makes sense with your above comment. Ran into a similar issue with Fleet for imported clusters having old namespaces get deleted after migration.